middle_end/flambda2/simplify_shared/slot_offsets.ml

```diff
@@ -428,27 +432,30 @@ module Greedy = struct
     slot.pos <- Assigned offset;
     List.iter (add_slot_offset_to_set offset slot) slot.sets;
     match slot.desc with
-    | Closure c ->
-      let (info : EO.closure_info) = EO.Id_slot { offset; size = slot.size } in
-      EO.add_closure_offset env c info
-    | Env_var v ->
-      let (info : EO.env_var_info) = EO.Var_slot { offset } in
-      EO.add_env_var_offset env v info
+    | Function_slot c ->
+      let (info : EO.function_slot_info) =
+        EO.Live_function_slot { offset; size = slot.size }
+      in
+      EO.add_function_slot_offset env c info
+    | Value_slot v ->
+      let (info : EO.value_slot_info) = EO.Live_value_slot { offset } in
+      EO.add_value_slot_offset env v info
 
   let mark_slot_as_removed env slot =
     match slot.pos with
     | Removed -> env
     | Unassigned -> (
       slot.pos <- Removed;
       match slot.desc with
-      | Closure _ -> Misc.fatal_error "Closure cannot be removed currently"
-      | Env_var v ->
-        let (info : EO.env_var_info) = EO.Dead_var in
-        EO.add_env_var_offset env v info)
+      | Function_slot _ ->
+        Misc.fatal_error "Function_slot cannot be removed currently"
```

Comment:
Note: we can actually remove function slots now, so this branch could be updated accordingly.

Reply:
I've added a CR